src/core/thread/topology.hpp

```diff
@@ -73,6 +73,9 @@ class Neighbor
      *
      */
     enum State
+#if _WIN32
```

Comment:
No, not unless all platforms we build on support this. I added a unit test case to show/validate the problem.